src/overlays/actors/ovl_Boss_07/z_boss_07.c

```diff
@@ -242,500 +636,7211 @@ ActorProfile Boss_07_Profile = {
+            velocity.z = Rand_CenteredFloat(5.0f);
+
+            accel.x = accel.z = 0.0f;
+            accel.y = -0.1f;
+
+            pos.y = Rand_ZeroFloat(10.0f) + 3.0f;
+
+            if ((u32)spawnPos != MAJORAS_INCARNATION_DUST_SPAWN_POS_FEET) {
+                pos.x = this->actor.focus.pos.x + Rand_CenteredFloat(150.0f);
+                pos.z = this->actor.focus.pos.z + Rand_CenteredFloat(150.0f);
+            } else {
+                pos.z = this->incarnationFeetPos[i].z + Rand_CenteredFloat(20.0f);
+                pos.x = this->incarnationFeetPos[i].x + Rand_CenteredFloat(20.0f);
+            }
+
+            func_800B0EB0(play, &pos, &velocity, &accel, &sDustPrimColor, &sDustEnvColor,
+                          Rand_ZeroFloat(150.0f) + 350.0f, 10, Rand_ZeroFloat(5.0f) + 14.0f);
+        }
+    }
+}
+
+void Boss07_MovePlayerFromCenter(PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    if (SQXZ(player->actor.world.pos) < SQ(80.0f)) {
+        player->actor.world.pos.z = 90.0f;
+    }
+}
+
+void Boss07_Wrath_SpawnDustAtPos(PlayState* play, Vec3f* spawnPos, u8 count) {
+    static Color_RGBA8 sDustPrimColor = { 60, 50, 20, 255 };
+    static Color_RGBA8 sDustEnvColor = { 40, 30, 30, 255 };
+    u8 i;
+
+    for (i = 0; i < count; i++) {
+        Vec3f pos;
+        Vec3f velocity;
+        Vec3f accel;
+
+        velocity.x = Rand_CenteredFloat(3.0f);
+        velocity.y = Rand_ZeroFloat(2.0f) + 1.0f;
+        velocity.z = Rand_CenteredFloat(3.0f);
+
+        accel.x = accel.z = 0.0f;
+        accel.y = -0.1f;
+
+        pos.x = spawnPos->x + Rand_CenteredFloat(30.0f);
+        pos.y = spawnPos->y + 15.0f + Rand_CenteredFloat(30.0f);
+        pos.z = spawnPos->z + Rand_CenteredFloat(30.0f);
+
+        func_800B0EB0(play, &pos, &velocity, &accel, &sDustPrimColor, &sDustEnvColor, Rand_ZeroFloat(50.0f) + 100.0f,
+                      10, Rand_ZeroFloat(5.0f) + 14.0f);
+    }
+}
+
+void Boss07_Wrath_ChooseJump(Boss07* this, PlayState* play, u8 canCancelCurrentJump) {
+    Player* player = GET_PLAYER(play);
+
+    if ((this->damagedTimer == 0) &&
+        (((this->actionFunc != Boss07_Wrath_Flip) && (this->actionFunc != Boss07_Wrath_StartJump) &&
+          (this->actionFunc != Boss07_Wrath_Jump)) ||
+         canCancelCurrentJump)) {
+        if (Rand_ZeroOne() < 0.5f) {
+            Boss07_Wrath_SetupFlip(this, play);
+        } else {
+            Boss07_Wrath_SetupJump(this, play);
+        }
+
+        this->disableCollisionTimer = 10;
+        this->whipWrapEndOffset = 0;
+
+        if (&this->actor == player->actor.parent) {
+            player->av2.actionVar2 = 101;
+            player->actor.parent = NULL;
+            player->csAction = PLAYER_CSACTION_NONE;
+        }
+    }
+}
+
+void Boss07_Wrath_JumpAwayFromExplosive(Boss07* this, PlayState* play) {
+    Actor* explosive = play->actorCtx.actorLists[ACTORCAT_EXPLOSIVES].first;
+
+    while (explosive != NULL) {
```

Comment:
Can this and similar loops be written as a for loop?

Reply:
All of them in Majora and most of them in Odolwa can. However, `Boss01_Bug_Crawl`, and writing this kind of thing with for-loops is actually really uncommon in the codebase right now, so I can't tell if I don't actually like it or it's just unfamiliar to me. You can see the change here; if people dislike it, I can revert the commit: https://github.com/zeldaret/mm/pull/1753/commits/e03e229893deb07f7dd821adb35e14c64411eac9